chromadb/test/property/strategies.py

```diff
@@ -458,17 +458,26 @@ def recordsets(
     num_unique_metadata: Optional[int] = None,
     min_metadata_size: int = 0,
     max_metadata_size: Optional[int] = None,
+    # ids can only be optional for add operations
+    for_add: bool = False,
 ) -> RecordSet:
     collection = draw(collection_strategy)
-
     ids = list(
         draw(st.lists(id_strategy, min_size=min_size, max_size=max_size, unique=True))
     )
 
+    # This probablistic event is used to mimic user behavior when they don't provide ids
+    if for_add and np.random.rand() < 0.5:
```

Comment:
its preferred not to use np.random.rand() in strategies, please use draw(st.bools)